### PR TITLE
python3Packages.configargparse: 1.3 -> 1.4

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 1.4

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 116540` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>27 packages built:</summary>
  <ul>
    <li>acpilight</li>
    <li>bashup-events32</li>
    <li>bashup-events44</li>
    <li>certbot (letsencrypt ,python38Packages.certbot)</li>
    <li>certbot-full</li>
    <li>coursera-dl</li>
    <li>gixy</li>
    <li>keepkey_agent (python38Packages.keepkey_agent)</li>
    <li>ledger_agent (python38Packages.ledger_agent)</li>
    <li>linkchecker</li>
    <li>python38Packages.ConfigArgParse</li>
    <li>python38Packages.certbot-dns-cloudflare</li>
    <li>python38Packages.certbot-dns-rfc2136</li>
    <li>python38Packages.certbot-dns-route53</li>
    <li>python38Packages.libagent</li>
    <li>trezor_agent (python38Packages.trezor_agent)</li>
    <li>xml2rfc (python38Packages.xml2rfc)</li>
    <li>python39Packages.ConfigArgParse</li>
    <li>python39Packages.certbot</li>
    <li>python39Packages.certbot-dns-cloudflare</li>
    <li>python39Packages.certbot-dns-rfc2136</li>
    <li>python39Packages.certbot-dns-route53</li>
    <li>resholve</li>
    <li>rofimoji</li>
    <li>snakemake</li>
    <li>styx</li>
    <li>yle-dl</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 116540` at ef75c33f run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>26 packages built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/9daef9db9c1d0f5af50b6dfbf81918ed">acpilight</a></li></li>
    <li><a href="https://gist.github.com/2f6279bf16e3c370e8c99c6f8bd99df8">bashup-events32</a></li></li>
    <li><a href="https://gist.github.com/c01a49a59c834e984c7387bdde8d9957">bashup-events44</a></li></li>
    <li><a href="https://gist.github.com/b35d385a5ecf1fe24ae8cf5604c0eab2">certbot</a></li> (python38Packages.certbot)</li>
    <li><a href="https://gist.github.com/0d2fd7bb455fc970672e3781388f08a0">coursera-dl</a></li></li>
    <li><a href="https://gist.github.com/b672c34137d38a9eb18183774ac48f5c">gixy</a></li></li>
    <li><a href="https://gist.github.com/0435513f39609b11f42abe035b57e658">keepkey_agent</a></li> (python38Packages.keepkey_agent)</li>
    <li><a href="https://gist.github.com/7b8bdabfe378c36dea05aa690919e577">ledger_agent</a></li> (python38Packages.ledger_agent)</li>
    <li><a href="https://gist.github.com/0e51270398629ace5d79167b98b35b75">linkchecker</a></li></li>
    <li><a href="https://gist.github.com/18a044d6ea31456ea38b30ff7a63cc30">python38Packages.ConfigArgParse</a></li></li>
    <li><a href="https://gist.github.com/529669112d03c2f0139be2ec901beec2">python38Packages.certbot-dns-cloudflare</a></li></li>
    <li><a href="https://gist.github.com/0f7cc6c7a8c35cbbb96cc28f9d1fb4dc">python38Packages.certbot-dns-rfc2136</a></li></li>
    <li><a href="https://gist.github.com/1fc1f456ff38672322987a76b539b6d3">python38Packages.certbot-dns-route53</a></li></li>
    <li><a href="https://gist.github.com/ed39ca99a767c8b673dd13ee19f69365">python38Packages.libagent</a></li></li>
    <li><a href="https://gist.github.com/9f5201816c06cab07a412254519f6637">trezor_agent</a></li> (python38Packages.trezor_agent)</li>
    <li><a href="https://gist.github.com/4049f5d138a8399a7fa638c1eda064fb">xml2rfc</a></li> (python38Packages.xml2rfc)</li>
    <li><a href="https://gist.github.com/b2441cddf3efd27cc89d98c64847f0d4">python39Packages.ConfigArgParse</a></li></li>
    <li><a href="https://gist.github.com/f7a2a6bda23297ca92a317615c45d09c">python39Packages.certbot</a></li></li>
    <li><a href="https://gist.github.com/02d8e99d76728f46f372d7d4e8f23978">python39Packages.certbot-dns-cloudflare</a></li></li>
    <li><a href="https://gist.github.com/115068d7f9ab879f2165e52993007b92">python39Packages.certbot-dns-rfc2136</a></li></li>
    <li><a href="https://gist.github.com/38a6688013989c6e5e702651331fa2a3">python39Packages.certbot-dns-route53</a></li></li>
    <li><a href="https://gist.github.com/91e116ad1d9d04052ee30fc268f2652b">resholve</a></li></li>
    <li><a href="https://gist.github.com/d946c7825dbbf3cb380ba7b6828b444a">rofimoji</a></li></li>
    <li><a href="https://gist.github.com/2482da6f95ab2349dbb7f13ded7d494f">snakemake</a></li></li>
    <li><a href="https://gist.github.com/ed60bfcf2f2407160ee12b489bdc78d6">styx</a></li></li>
    <li><a href="https://gist.github.com/6d62775b4f2719b7baed46e88c8706cb">yle-dl</a></li></li>
  </ul>
</details>

File: pkgs/development/python-modules/configargparse/default.nix [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 buildPythonPackage rec {
   pname = "ConfigArgParse";
-  version = "1.3";
+  version = "1.4";
 
   src = fetchFromGitHub {
     owner = "bw2";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/configargparse/default.nix:13]: You forgot to update the hash.

Comment (fabaff) [reply pkgs/development/python-modules/configargparse/default.nix:13]: Thanks. Fixed.

The package now follows PEP 0503.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 116540` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>27 packages built:</summary>
  <ul>
    <li>acpilight</li>
    <li>bashup-events32</li>
    <li>bashup-events44</li>
    <li>certbot (letsencrypt ,python38Packages.certbot)</li>
    <li>certbot-full</li>
    <li>coursera-dl</li>
    <li>gixy</li>
    <li>keepkey_agent (python38Packages.keepkey_agent)</li>
    <li>ledger_agent (python38Packages.ledger_agent)</li>
    <li>linkchecker</li>
    <li>python38Packages.ConfigArgParse</li>
    <li>python38Packages.certbot-dns-cloudflare</li>
    <li>python38Packages.certbot-dns-rfc2136</li>
    <li>python38Packages.certbot-dns-route53</li>
    <li>python38Packages.libagent</li>
    <li>trezor_agent (python38Packages.trezor_agent)</li>
    <li>xml2rfc (python38Packages.xml2rfc)</li>
    <li>python39Packages.ConfigArgParse</li>
    <li>python39Packages.certbot</li>
    <li>python39Packages.certbot-dns-cloudflare</li>
    <li>python39Packages.certbot-dns-rfc2136</li>
    <li>python39Packages.certbot-dns-route53</li>
    <li>resholve</li>
    <li>rofimoji</li>
    <li>snakemake</li>
    <li>styx</li>
    <li>yle-dl</li>
  </ul>
</details>

**Pull request merged**